src/main/java/net/sf/jabref/logic/citationstyle/CitationStyleGenerator.java

```diff
@@ -0,0 +1,74 @@
+package net.sf.jabref.logic.citationstyle;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import net.sf.jabref.logic.formatter.bibtexfields.UnicodeToLatexFormatter;
+import net.sf.jabref.logic.l10n.Localization;
+import net.sf.jabref.model.entry.BibEntry;
+
+import de.undercouch.citeproc.CSL;
+import de.undercouch.citeproc.bibtex.BibTeXConverter;
+import de.undercouch.citeproc.bibtex.BibTeXItemDataProvider;
+import de.undercouch.citeproc.output.Bibliography;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.jbibtex.BibTeXDatabase;
+import org.jbibtex.ParseException;
+import org.jbibtex.TokenMgrError;
+
+
+/**
+ * WARNING: The generation of a citation may take some time, better call it in outside the main Thread
+ */
+public class CitationStyleGenerator {
```

Comment:
I don't like package-private (because writing nothing looks like somebody just forgot to specify the correct visibility)